src/adapters/auth-adapters.ts

```diff
@@ -0,0 +1,118 @@
+import {getCookie , setCookie, deleteCookie} from '../util/Authentication'
+
+const API_URL = 'http://localhost:8000/api/v1'
+let endpoint = ''
+export const fetchAPIData = async (url: string, method: string, body:object , headers?:object) => {
+    const token = getCookie('shelf-share-session')
+    console.log('token',token)
+    const data = await fetch(API_URL + endpoint , {
+        method,
+        body: JSON.stringify(body),
+        headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+            ...headers,
+        },
+    })
+    const response = await data.json();
+    console.log('res', response)
+    setCookie('shelf-share-session', JSON.stringify(body) , 1 )
+    return response; 
+}
+
+
+//signUp
+type SignUpInput = {
+    email: string,
+    password: string,
+    username: string,
+    givenName: string,
+    familyName: string,
+    dateOfBirth: Date,
+    address: string,
+    role: string,
+    latitude: number,
+    longitude: number,
+}
+/**
+* This function signs up a user with the provided information.
+* @param {Object} SignUpInputs - The input object containing user's signup information.
+* @param {string} SignUpInputs.email - User's email address.
+* @param {string} SignUpInputs.password - User's password.
+* @param {string} SignUpInputs.givenName - User's given name.
+* @param {string} SignUpInputs.username - User's username.
+* @param {date} SignUpInputs.dateOfBirth - User's date of birth.
+* @param {string} SignUpInputs.familyName - User's family name.
+* @param {string} SignUpInputs.address - User's address.
+* @param {'admin'|'user'} SignUpInputs.role - User's role.
+* @param {number} SignUpInputs.latitude - User's latitude.
+* @param {number} SignUpInputs.longitude - User's longitude.
+* @example
+* const SignUpInputs = {
+*   email: "john@gmail.com",
+*   password: "secretpassword1234",
+*   username: "John",
+*   givenName: "JohnAB",
+*   familyName: "ABC",
+*   dateOfBirth: new Date("1991-12-02"),
+*   address: "6015 Farrington Ave, Alexandria, VA  22304, United States",
+*   role: 'user',
+*   latitude: 38.797667,
+*   longitude: -77.14066,
+* };
+* export const signUpAdapter = async (SignUpInputs) => {
+*   // function implementation here
+* };
+* @returns {Promise<Object>} A promise that resolves to the user's signup information.
+*/
+export const signUpAdapter = async (signUpInput: SignUpInput) =>{
+    endpoint = `/user/sign-up`
+    //console.log('tesst', url)
+    const data = await fetchAPIData(endpoint, 'POST', signUpInput)
+    setCookie('shelf-share-session', 'data', 1)
```

Comment:
Data should be the actual data value rather than a string. You will also want to stringify the data into JSON string using JSON.stringify(data). You should verify that this works by calling this function and checking that the cookie is set to the userobject.

Reply:
Jacob, as you mentioned I check the cookie at my browser application part cookie. cookie is existed with a message which says please write your email address and password for loginAdopter function. So I am getting 400 error because I need to have email address and password. in this case I am thinking that my code is working fine!

Reply:
I would expect a 400 if you enter no email and password. You should test with entering a correct password as well. In this case, the cookie should include the user data.